### PR TITLE
Fix for issue when OAuth scope creation gives 500 error code for invalid request body 

### PR DESCRIPTION
**Pull request by mevan-karu**
Issue : https://github.com/wso2/product-is/issues/7615

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -460,6 +460,12 @@
             </dependency>
 
             <!--Swagger Dependencies-->
+            <dependency>
+                <groupId>com.fasterxml.jackson.core</groupId>
+                <artifactId>jackson-core</artifactId>
+                <scope>provided</scope>
```
Comment (IsuraD) [inline pom.xml]: scope is not required here

File: components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/exceptionMapper/JsonProcessingExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    public Response toResponse(UnrecognizedPropertyException e) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Provided JSON request content is not in the valid format:", e);
+        }
+
+        ErrorDTO errorDTO = new ErrorDTO();
+        errorDTO.setCode(Response.Status.BAD_REQUEST.toString());
+        errorDTO.setMessage(ERROR_MESSAGE);
+        errorDTO.setDescription(DESCRIPTION);
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/exceptionMapper/JsonProcessingExceptionMapper.java]: Shall we put the actual description in the error message? That would be more meaningful

File: pom.xml (hunk 1 above)
Comment (mevan-karu) [reply pom.xml]: Fixed

File: components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/webapp/WEB-INF/beans.xml [hunk 3]
```diff
@@ -13,6 +13,7 @@
         </jaxrs:serviceBeans>
         <jaxrs:providers>
             <bean class="com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider"/>
+            <bean class="org.wso2.carbon.identity.oauth.scope.endpoint.expmapper.JsonProcessingExceptionMapper"/>
         </jaxrs:providers>
     </jaxrs:server>
 </beans>
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/webapp/WEB-INF/beans.xml]: Add a new line EOF

File: components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/expmapper/JsonProcessingExceptionMapper.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import javax.ws.rs.ext.ExceptionMapper;
+
+/**
+ * Handles exceptions when an incorrect json requests body is received.
+ * Sends a default error response.
+ */
+public class JsonProcessingExceptionMapper implements ExceptionMapper<UnrecognizedPropertyException> {
+
+    private static final Log log = LogFactory.getLog(JsonProcessingExceptionMapper.class);
+    public static final String ERROR_MESSAGE = "Invalid Request";
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/expmapper/JsonProcessingExceptionMapper.java]: Shall we move the constants to Oauth2ScopeConstants.java file

Comment (mevan-karu) [reply components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/java/org/wso2/carbon/identity/oauth/scope/endpoint/expmapper/JsonProcessingExceptionMapper.java]: Added a new error message to Oauth2ScopeConstants.java

File: components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/webapp/WEB-INF/beans.xml (hunk 3 above)
Comment (mevan-karu) [reply components/org.wso2.carbon.identity.oauth.scope.endpoint/src/main/webapp/WEB-INF/beans.xml]: Added

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/Oauth2ScopeConstants.java [hunk 5]
```diff
@@ -42,6 +42,7 @@
                 "Scope with the name %s already exists in the system. Please use a different scope name."),
         ERROR_CODE_BAD_REQUEST_SCOPE_NOT_SPECIFIED("41005", "Scope is not specified."),
         ERROR_CODE_BAD_REQUEST_SCOPE_NAME_CONTAINS_WHITESPACES("41006", "Scope name: %s contains white spaces."),
+        ERROR_CODE_BAD_REQUEST("41007", "Invalid request"),
```
Comment (Buddhimah) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/Oauth2ScopeConstants.java:45]: Let's add a full stop

**Pull request merged**